ci/compute-matrix.jq

```diff
@@ -14,6 +14,9 @@ def compute_ubuntu_version($x):
 def compute_cuda_tag($x):
   $x + {CUDA_TAG: $x.CUDA_VER | split(".") | [.[0], .[1]] | join(".") };
 
+def compute_build_raft_ann_bench_cpu_image($x):
+  $x + {BUILD_RAFT_ANN_BENCH_CPU_IMAGE: ($x.CUDA_VER == "12.5.1")}; # we don't need to build CPU packages for different CUDA versions.
```

Comment:
Thanks for single-sourcing this, I like this approach of handling it here along with the other matrix operations.

Is it possible to go even further and make this condition something like "sort all options by `CUDA_VER` and take the latest CUDA version"?

That way, as `12.5.1` drops out of the matrix in the future, this would continue to work without modifications required.

My `jq` is not very good so I can't recommend the exact syntax to use here, but I know we've done something similar in other CI configs like this:

```yaml
# Package is pure Python and only ever requires one build.
# This selects "ARCH=amd64 + the latest supported Python + CUDA".
matrix_filter: map(select(.ARCH == "amd64")) | max_by([(.PY_VER|split(".")|map(tonumber)), (.CUDA_VER|split(".")|map(tonumber))]) | [.]
```

https://github.com/rapidsai/dask-cuda/blob/00c37dc55bee1a34f7d9f6599a89a3f89c15651b/.github/workflows/pr.yaml#L52-L54